### PR TITLE
Added ability to add Custom Emojis

### PR DESCRIPTION
**Pull request by ricking06**
customized emojis .....

### CODE REVIEW

File: src/components/picker.js [hunk 1]
```diff
@@ -290,7 +296,95 @@ export default class Picker extends React.Component {
...
+           emojis : elist,
+           custom : true,
+           emojis_src : esrc,
+           anchor : false
+         };
+         data.categories.unshift(category_custom);
+     }
+   this.newdata = data;
+   self.initializeCategories();
+   console.log(data);
```
Comment (ricking06) [inline src/components/picker.js]: remove this line

File: src/components/search.js [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+    this.state = {
+      placeholder : null
+    };
+  }
   handleChange() {
     var { input } = this.refs,
-        value = input.value
+        value = input.value;
 
+    value = ltrim(rtrim(value,':'),':');
```
Comment (ricking06) [inline src/components/search.js]: privously wasnt able to search starting with `:`  so added as default functionality. removed `:` from both sides

File: src/utils/index.js [hunk 3]
```diff
@@ -6,18 +6,25 @@ const SKINS = [
   '1F3FD', '1F3FE', '1F3FF',
 ]
 
-function unifiedToNative(unified) {
+function unifiedToNative(unified,custom) {
```
Comment (ricking06) [inline src/utils/index.js]: i dont exactly understand that unified is.. changed `unified  = name` for custom emojis... pre-loaded emojis are unnaffected by this.

File: src/components/picker.js (hunk 1 above)
Comment (ricking06) [inline src/components/picker.js]: console.log

File: src/components/picker.js [hunk 4]
```diff
@@ -133,15 +138,17 @@ export default class Picker extends React.Component {
   }
 
   handleEmojiOver(emoji) {
-    var { preview } = this.refs
+    var { preview,search } = this.refs
     preview.setState({ emoji: emoji })
+    search.setState({placeholder : emoji.colons })
```
Comment (ricking06) [inline src/components/picker.js]: now onMouseover emoji  the  colon will be displayed as placeholder on searchbar.

**Comment by EtienneLem** [comment]
Hey mate!

Happy to see you’re making progress on your fork. I’m afraid that I cannot merge as-is because the PR is tracking your master branch and includes a lot of changes unrelated to custom emojis and it seems a little bit too complex to isolate just the custom emojis part from the rest. I also see some styles change that I don’t think people using EmojiMart would expect.

I would say a fork is definitely the way to go for your use case. I see you’ve published your version on NPM, I shall direct people to your version if they have similar needs. 🤘

**Pull request closed without merging**